openhands/core/message.py

```diff
@@ -80,17 +80,32 @@ def serialize_model(self) -> dict:
             elif isinstance(item, ImageContent) and self.vision_enabled:
                 content.extend(d)
 
+        # For DeepSeek, we need to ensure content is a string when using tool calls
```

Comment:
For DeepSeek we need to ensure content is a string. But this change in message.py applies to all LLMs supported by liteLLM! That is a serious problem. We need to make the content into a single string ONLY for some LLMs:
-  that don't support lists of dicts in content, but require a string instead.

This is important! The current function calling implementation uses the format of Anthropic Claude variants. Think again what we need to do in order to keep it working for Anthropic, and implement it differently for DeepSeek, while NOT breaking everything else.

This can help:
- verify how we define SUPPORT_PROMPT_CACHE_MODELS in llm.py
- verify how we use such module globals in llm.py
- DeepSeek might work with function calling, if it has the right format with a single string for content, at least for system message. Make a special case this time, so that you don't break everything else.
- there are LLMs that don't work with prompt caching, nor with vision, nor with function calling. For those, make a second serialization algo, to make their contents into one content string per message (not a list of contents).